Developers/micropayments.md

```diff
@@ -0,0 +1,130 @@
+# Trustless micropayment channels
+
+## Protocol
+
+![Micropayment Protocol](micropayment_protocol.png)
+
+
+### Deposit Transaction
+
+The deposit transaction ensures the payee cannot spend the funds while
+the channel is open.
+
+The pay to script allows for three possible outcoms:
+
+ - The payer can recover the deposit after a timeout if no commit was made.
+ - Both parties commit and close the channel.
+ - The payer collects the change after the payee publishes the spend secret.
+
+**The commit transaction spend secret must be the same as the deposit
+transaction spend secret!**
+
+pay to script:
+
+    OP_IF
+        2 <payer pubkey> <payee pubkey> 2 OP_CHECKMULTISIG
+    OP_ELSE
+        OP_IF
+            OP_HASH160 <spend secret hash> OP_EQUALVERIFY
+            <payer pubkey> OP_CHECKSIG
+        OP_ELSE
+            <expire time> OP_CHECKSEQUENCEVERIFY OP_DROP
+            <payer pubkey> OP_CHECKSIG
+        OP_ENDIF
+    OP_ENDIF
+
+
+### Expire Transaction
+
+The expire transaction is used by the payee to recover the and funds after
+the channel expires. This is to prevent funds being lost should the payee
+be hit by a bus.
+
+script sig:
+
+    <payer signature> OP_FALSE OP_FALSE
+
+    # Example: 796637e788db8d3430ae48ec057ef493a871b7abc4711f9bb5f6148efc2212e7
+
+
+### Change Transaction
+
+After the commit transaction has been published on the blockchain, the
+payee must reveal the spend secret to spend the committed funds. The payer
+can then recover the change from the channel before it times out by using the
+revealed spend secret.
+
+script sig:
+
+    <payer signature> <spend secret> OP_TRUE OP_FALSE
+
+    # Example: a6fdd23b2e5b1153cbd358c362456263531451b80ab4a597f7aa122cecc52703
+
+
+### Commit Transaction
+
+The commit transaction is used to transfer funds and prevent a race condition
+between the payee recovering the payout and the payer recovering the change.
+
+Every time the payer wishes to transfer funds to the payee, the output amount
+of the transaction is increased. The payee then signs and shares it.
```

Comment:
the payer signs and shares the commit TX, not the payee right? your diagram shows it that way at least

and otherwise I wouldnt know how the payee is protected against the payer just broadcasting the first version of the commit TX of the series?